File: app/src/main/java/com/travlog/android/apps/viewmodels/outputs/NoteViewModelOutputs.java
package com.travlog.android.apps.viewmodels.outputs;

import android.support.annotation.NonNull;

import com.travlog.android.apps.models.Note;

import io.reactivex.Completable;
import io.reactivex.Observable;

public interface NoteViewModelOutputs {

    @NonNull
    Observable<String> setTitleText();

    @NonNull
    Observable<Note> showEditNoteActivity();

    @NonNull
    Completable back();
}

File: app/src/main/java/com/travlog/android/apps/viewmodels/outputs/NoteViewModelOutputs.kt
package com.travlog.android.apps.viewmodels.outputs

import com.travlog.android.apps.models.Destination
import com.travlog.android.apps.models.Note

import io.reactivex.Completable
import io.reactivex.Observable

interface NoteViewModelOutputs {

    fun setTitleText(): Observable<String>

    fun updateDestinations(): Observable<List<Destination>>

    fun showEditNoteActivity(): Observable<Note>

    fun finish(): Completable
}
